pandas/tests/series/test_missing.py

```diff
@@ -297,6 +297,11 @@ def test_fillna_raise(self):
         self.assertRaises(TypeError, s.fillna, [1, 2])
         self.assertRaises(TypeError, s.fillna, (1, 2))
 
+        # related GH 9217, make sure limit is greater than 0
+        for limit in [-1, 0]:
+            s = Series([1, 2, 3, None])
+            tm.assertRaises(AssertionError, lambda: s.fillna(1, limit=limit))
```

Comment:
we always raise ValueError, never AssertionError (which is an internal error and should not be user visible)